scripts/machine_translation/train_gnmt.py

```diff
@@ -190,43 +214,94 @@ def evaluate(data_loader):
     return avg_loss, real_translation_out
 
 
+def clip_global_norm(arrays, max_norm, context_len=len(context)):
+    """Enable the clip_global_norm function at multi contexts.
+       Reference to gluon.utils.clip_global_norm()
+
+    """
+    def _norm(array):
+        if array.stype == 'default':
+            x = array.reshape((-1,))
+            return ndarray.dot(x, x)
+        return array.norm().square()
+
+    assert len(arrays) > 0
+    assert context_len >= 1
+
+    ctx = arrays[0].context
+    arrays_at_all_context = arrays[0]
+    # multi GPUs need to sum up each row(parameters at each context) first
+    if context_len > 1:
+        for i in range(context_len):
+            arrays_at_all_context += arrays[i]
```

Comment:
What is this here? Sum up the first context_len elements?

Reply:
This code is mean to sum up the same parameter at different contexts. For these parameters is Ndarray type, so i use `+=` derictly